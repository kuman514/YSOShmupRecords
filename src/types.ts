export interface NavNodeInfo {
  id: string;
  label: string;
  childNavNodeIds: NavNodeInfo['id'][];
}

export interface ShmupRecord {
  id: string;
  when: Date;
  subjectId: string;
  stage: string;
  /**
   * Why score should be string?
   * In case of exceeding `Number.MAX_SAFE_INTEGER`, which equals to `9007199254740991`.
   */
  score: string;
  byWhat: string;
  comment: string;
  thumbnailUrl: string;
  originalImageUrl: string;
  tweetUrl: string;
  youtubeUrl?: string;
  specialTags?: string[];
}
