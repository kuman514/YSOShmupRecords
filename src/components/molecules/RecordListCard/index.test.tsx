import { cleanup, render, screen } from '@testing-library/react';
import 'jest-styled-components';
import { beforeEach, describe, expect, it } from 'vitest';

import AvatarJpgUrl from '^/assets/avatar/avatar.jpeg';
import { ShmupRecordPreview } from '^/types';

import { RecordListCard } from '.';

const previewForTest: ShmupRecordPreview = {
  title: 'kuman514',
  recordId: 'test-test',
  when: new Date('2024-01-01'),
  typeId: 'test',
  stage: '2-5',
  score: '12345679',
  byWhat: 'test',
  comment: 'test',
  thumbnailUrl: AvatarJpgUrl,
  originalImageUrls: [],
};

describe('RecordListCard', () => {
  beforeEach(() => {
    cleanup();
  });

  it('should have a snapshot match for its appearance', () => {
    const { container } = render(
      <RecordListCard recordPreview={previewForTest} />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByText(/kuman514/i)).not.toBeNull();
  });

  it('should have a snapshot match showing youtube mark and special tags', () => {
    const { container } = render(
      <RecordListCard
        recordPreview={{
          ...previewForTest,
          youtubeUrl: 'youtube url',
          specialTags: ['yasuo', 'koishi', 'no-miss-all'],
        }}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
