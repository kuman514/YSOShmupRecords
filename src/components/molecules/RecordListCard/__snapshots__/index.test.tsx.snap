// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RecordListCard > should have a snapshot match for its appearance 1`] = `
.c0 {
  position: relative;
  width: 300px;
  min-height: 240px;
  display: grid;
  grid-template-rows: 200px 1fr;
  background-color: var(--primary-color);
  border-radius: 16px;
  overflow: hidden;
}

.c0:hover {
  background-color: var(--hovering-color);
}

.c0:hover>div>img {
  transform: scale(1.3);
}

.c0:hover>div>img+div {
  opacity: 1;
}

.c1 {
  width: 100%;
  height: 100%;
  overflow: hidden;
  border-top-left-radius: 16px;
  border-top-right-radius: 16px;
}

.c2 {
  width: 100%;
  height: 100%;
  object-fit: cover;
  transition: transform 100ms ease-in-out;
}

.c3 {
  width: 100%;
  height: 200px;
  position: absolute;
  left: 0;
  top: 0;
  background-color: rgba(255, 255, 255, 0.3);
  opacity: 0;
  transition: opacity 100ms ease-in-out;
}

.c4 {
  height: 100%;
  display: flex;
  flex-direction: column;
  row-gap: 8px;
  padding: 8px 16px;
  justify-content: center;
  align-items: flex-start;
}

.c5 {
  color: var(--white-color);
  font-size: 16px;
  font-weight: 600;
}

.c6 {
  color: var(--white-color);
  font-size: 16px;
  font-weight: 400;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="kumankoishi shmup record 2024-01-01"
      class="c2"
      src="/src/assets/avatar/avatar.jpeg"
    />
    <div
      class="c3"
    />
  </div>
  <div
    class="c4"
  >
    <span
      class="c5"
    >
      kuman514
    </span>
    <span
      class="c5"
    >
      kumankoishi shmup record 2024-01-01
    </span>
    <span
      class="c6"
    >
      2-5 / 12345679점
    </span>
  </div>
</div>
`;

exports[`RecordListCard > should have a snapshot match showing youtube mark and special tags 1`] = `
.c8 {
  padding: 4px 8px;
  justify-content: center;
  align-items: center;
  background-color: var(--gray-color);
  border-radius: 4px;
  font-size: 12px;
  font-weight: 400;
  color: var(--white-color);
}

.c0 {
  position: relative;
  width: 300px;
  min-height: 240px;
  display: grid;
  grid-template-rows: 200px 1fr;
  background-color: var(--primary-color);
  border-radius: 16px;
  overflow: hidden;
}

.c0:hover {
  background-color: var(--hovering-color);
}

.c0:hover>div>img {
  transform: scale(1.3);
}

.c0:hover>div>img+div {
  opacity: 1;
}

.c1 {
  width: 100%;
  height: 100%;
  overflow: hidden;
  border-top-left-radius: 16px;
  border-top-right-radius: 16px;
}

.c2 {
  width: 100%;
  height: 100%;
  object-fit: cover;
  transition: transform 100ms ease-in-out;
}

.c3 {
  width: 100%;
  height: 200px;
  position: absolute;
  left: 0;
  top: 0;
  background-color: rgba(255, 255, 255, 0.3);
  opacity: 0;
  transition: opacity 100ms ease-in-out;
}

.c4 {
  height: 100%;
  display: flex;
  flex-direction: column;
  row-gap: 8px;
  padding: 8px 16px;
  justify-content: center;
  align-items: flex-start;
}

.c5 {
  color: var(--white-color);
  font-size: 16px;
  font-weight: 600;
}

.c6 {
  color: var(--white-color);
  font-size: 16px;
  font-weight: 400;
}

.c7 {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  gap: 8px;
}

.c9 {
  position: absolute;
  top: 16px;
  right: 16px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="kumankoishi shmup record 2024-01-01"
      class="c2"
      src="/src/assets/avatar/avatar.jpeg"
    />
    <div
      class="c3"
    />
  </div>
  <div
    class="c4"
  >
    <span
      class="c5"
    >
      kuman514
    </span>
    <span
      class="c5"
    >
      kumankoishi shmup record 2024-01-01
    </span>
    <span
      class="c6"
    >
      2-5 / 12345679점
    </span>
    <div
      class="c7"
    >
      <span
        class="c8"
      >
        yasuo
      </span>
      <span
        class="c8"
      >
        koishi
      </span>
      <span
        class="c8"
      >
        노미스 ALL
      </span>
    </div>
  </div>
  <svg
    class="c9"
    height="33.6"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 28.57 20"
    width="48"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M27.9727 3.12324C27.6435 1.89323 26.6768 0.926623 25.4468 0.597366C23.2197 2.24288e-07 14.285 0 14.285 0C14.285 0 5.35042 2.24288e-07 3.12323 0.597366C1.89323 0.926623 0.926623 1.89323 0.597366 3.12324C2.24288e-07 5.35042 0 10 0 10C0 10 2.24288e-07 14.6496 0.597366 16.8768C0.926623 18.1068 1.89323 19.0734 3.12323 19.4026C5.35042 20 14.285 20 14.285 20C14.285 20 23.2197 20 25.4468 19.4026C26.6768 19.0734 27.6435 18.1068 27.9727 16.8768C28.5701 14.6496 28.5701 10 28.5701 10C28.5701 10 28.5677 5.35042 27.9727 3.12324Z"
        fill="#FF0000"
      />
      <path
        d="M11.4253 14.2854L18.8477 10.0004L11.4253 5.71533V14.2854Z"
        fill="white"
      />
    </g>
  </svg>
</div>
`;
